scripts/scripts.js

```diff
@@ -212,9 +213,162 @@ function decorateFeatImg(getMetadata) {
   if (featImgMeta === 'off') document.body.classList.add('hide-feat-img');
 }
 
+// new functions
+function decorateMediaBlock() {
+  const mediaBlocks = document.querySelectorAll('.media');
+  if (!mediaBlocks || mediaBlocks.length === 0) return;
+
+  mediaBlocks.forEach((block) => {
+    if (block.classList.contains('card-as-link')) {
+      const link = block.querySelector('a');
+      if (!link) console.error('missing link in Media card');
+
+      const cardLink = document.createElement('a');
+      cardLink.setAttribute('class', block.classList);
+      cardLink.href = link.href;
+      cardLink.target = link.target;
+
+      if (block.classList.contains('hide-link-text')) {
+        const linkParent = link.closest('p');
+        if (linkParent) {
+          linkParent.remove();
+        } else {
+          link.remove();
+        }
+      }
+
+      cardLink.innerHTML = block.innerHTML;
+
+      block.parentNode.replaceChild(cardLink, block);
+    }
+  });
+}
+
+function decorateMasonryBrick() {
+  const bricks = document.querySelectorAll('.masonry-layout .brick');
+  if (!bricks || bricks.length === 0) return;
+
+  bricks.forEach((brick) => {
+    const links = brick.querySelectorAll('a');
+    if (links && links.length === 1) {
+      const link = links[0];
+      brick.setAttribute('data-link', link.href);
+      brick.addEventListener('click', () => {
+        window.location.href = link.href;
+      });
+    }
+  });
+}
+
+function cloneBlockToSidebar(sidebar, blockClass) {
+  const targetBlock = document.querySelector(`.${blockClass}`);
+  if (!targetBlock) return false;
+
+  const clonedBlock = targetBlock.cloneNode(true);
+  targetBlock.classList.add('hide-if-sidebar-visible');
+
+  sidebar.append(clonedBlock);
+
+  return clonedBlock;
+}
+
+async function initSidebar() {
+  const sidebar = document.createElement('div');
+  sidebar.classList.add('blog-wrapper-sidebar');
+
+  const sidebarArticleMeta = cloneBlockToSidebar(sidebar, 'article-meta');
+  await initSharingLinkFunction(sidebarArticleMeta);
+  cloneBlockToSidebar(sidebar, 'banner');
+  cloneBlockToSidebar(sidebar, 'tags');
+
+  return sidebar;
+}
+
+function moveRecommendedArticleAsFinalSection(main) {
+  const recBlock = main.querySelector('.recommended-articles');
+  if (!recBlock) return;
+
+  const section = document.createElement('div');
+  section.classList.add('section');
+  section.append(recBlock);
+  main.append(section);
+}
+
 const { loadArea, setConfig, getMetadata } = await import(`${miloLibs}/utils/utils.js`);
 
+async function setUpSidebarLayoutForBlogPage() {
+  const main = document.querySelector('main');
+  const childContent = document.querySelector('.content');
+  const blogDetailUrlIdentifier = '/publish/';
+
+  const isBlogPage = getMetadata('content-type') === 'article' && window.location.pathname.includes(blogDetailUrlIdentifier);
```

Comment:
As per our other conversation. Using `getMetadata('content-type') === 'article'` should be enough, that's what we currently use to flag article pages.

Reply:
It probably makes more sense to name this variable `isArticlePage` since this is only applying to article pages. 

There are other types of pages under the blog that get different styling than article pages. E.g. topics, author, homepage, etc.

Reply:
Thanks for the insight & confirmation, just pushed the update